assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/fields/BaseNumberInputField.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useCallback } from 'react';
+import classNames from 'classnames';
+
+import { BaseFieldProps, FieldValue, InputProps } from './types';
+import { NumberInput } from '@infraUI/inputs';
+
+import useBaseField from './useBaseField';
+
+const BaseNumberInputField: React.FC<BaseFieldProps> = ({
+	children,
+	component,
+	disabled,
+	name,
+	format,
+	formatOnBlur,
+	parse,
+	placeholder,
+	getValue,
+	setValue,
+	value,
+	...props
+}) => {
+	const [fieldValue, setFieldValue] = React.useState(Number(value ?? getValue()));
+
+	const className = classNames(props.className, 'ee-input-base ee-input');
+	const onChange = useCallback(
+		(value) => {
+			setFieldValue(value);
+			setValue(parse(value, name));
+		},
+		[name, parse, setValue]
+	);
```

Comment:
Maintaining a local state may fail in case it's a base price and `reverseCalculate` is `true`. Changing of ticket total won't reflect the change in base price.

Reply:
updated: https://github.com/eventespresso/event-espresso-core/pull/2888/commits/d501db03e4fdb4d653b3e2cfcec8fa4c68ab53b8

Reply:
removed local state, had added this before because of that string to number coercion, but wrapped `fieldValue` with number and it's good to go